src/main/java/com/jfrog/ide/idea/scan/ScannerBase.java

```diff
@@ -219,7 +219,7 @@ private void visitDepTreeNode(Map<String, DependencyNode> dependencies, DepTree
                 }
                 DependencyNode existingDep = addedDeps.get(descriptorPath).get(compId);
                 if (existingDep != null) {
-                    existingDep.setIndirect(indirect);
+                    existingDep.setIndirect(indirect && existingDep.isIndirect());
```

Comment:
```suggestion
                   // Preserve previous "direct" information  
                    if(existingDep.isIndirect()){
                       existingDep.setIndirect(indirect);
                     }
```
I believe it is a bit more clear that way :)